test/e2e_node/resource_metrics_test.go

```diff
@@ -122,22 +105,32 @@ var _ = SIGDescribe("ResourceMetricsAPI", nodefeature.ResourceMetrics, func() {
 			gomega.Consistently(ctx, getResourceMetrics, 1*time.Minute, 15*time.Second).Should(matchResourceMetrics)
 		})
 		ginkgo.AfterEach(func(ctx context.Context) {
-			ginkgo.By("Deleting test pods")
-			var zero int64 = 0
-			e2epod.NewPodClient(f).DeleteSync(ctx, pod0, metav1.DeleteOptions{GracePeriodSeconds: &zero}, 10*time.Minute)
-			e2epod.NewPodClient(f).DeleteSync(ctx, pod1, metav1.DeleteOptions{GracePeriodSeconds: &zero}, 10*time.Minute)
-			if !ginkgo.CurrentSpecReport().Failed() {
-				return
-			}
-			if framework.TestContext.DumpLogsOnFailure {
-				e2ekubectl.LogFailedContainers(ctx, f.ClientSet, f.Namespace.Name, framework.Logf)
-			}
-			ginkgo.By("Recording processes in system cgroups")
-			recordSystemCgroupProcesses(ctx)
+			removeMetricsPods(ctx, f)
 		})
 	})
 })
 
+func createMetricsPods(ctx context.Context, f *framework.Framework) {
+	ginkgo.By("Creating test pods to measure their resource usage")
+	numRestarts := int32(1)
+	pods := getSummaryTestPods(f, numRestarts, pod0, pod1)
+	e2epod.NewPodClient(f).CreateBatch(ctx, pods)
+
+	ginkgo.By("Restarting the containers to ensure container metrics are still being gathered after a container is restarted")
+	gomega.Eventually(ctx, func(ctx context.Context) error {
+		for _, pod := range pods {
+			err := verifyPodRestartCount(ctx, f, pod.Name, len(pod.Spec.Containers), numRestarts)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	}, time.Minute, 5*time.Second).Should(gomega.Succeed())
+
+	ginkgo.By("Waiting 15 seconds for cAdvisor to collect 2 stats points")
+	time.Sleep(15 * time.Second)
```

Comment:
tests that are using this method has it's own `Eventually` call that waits for a minute. Do we need these `15` seconds here?